### ISSUE TITLE
Question about assertion in calculation of two-electron integrals

### ISSUE THREAD

**Issue by alxvov** [pyscf/pyscf#1089]
Hi all,

I am trying to calculate the 2 electron integrals for finite k-point sampling. I follow the example given in pbc folder. My system has 4 electrons  and 6 basis functions, and I am able to run the code when my k-point sampling is 2x2x2. However, when I increase the k-point sampling up to 3x3x3 or more, I get the following assertion error:

  File "run.py", line 103, in eris2h2
    eri_kpt = kmf.with_df.ao2mo([kmf.mo_coeff[i] for i in kptspqrs], [kpts[i] for i in kptspqrs])
  File "/ScratchArray/home/aleksei/pyscf/venv/lib/python3.7/site-packages/pyscf/pbc/df/fft_ao2mo.py", line 150, in general
    eri = _contract_plain(mydf, mos, coulG, fac, max_memory=max_memory)
  File "/ScratchArray/home/aleksei/pyscf/venv/lib/python3.7/site-packages/pyscf/pbc/df/fft_ao2mo.py", line 196, in _contract_plain
    assert blksize > 0
AssertionError

Do you possible know what I need to do in this case? Do I have to increase the max memory? I tried to do this:

kmf = scf.KRHF(cell, kpts)
kmf.max_memory = 16000

but it seems that it doesn't help. I also tried to use different number of threads 1, 4, 8, 16 but it didn't help either. Any advice? 

Many thanks,
Aleksei

**Comment by sunqm** [comment]
I believe this is caused by the energy cutoff. This test system may require a high energy cutoff, which make FFT run out of memory. What is the value of cell.ke_cutoff?

**Comment by alxvov** [comment]
Hello @jamesETsmith and @sunqm, thank you both for your quick reply. Working out minimal working example helped me to understand what's the problem. The problem was that I stored all the integrals in the memory and even for this 'small' system this requires around 11 GB. The kinetic energy cutoff is around 900 eV (216000 coefficients) which is not too bad I guess. At first, it was not clear to me that this assertion I get is due to memory limits because I tried to use 16 GB but that was not enough. Increasing it futher (I bassically reqested all the memory I have on the node) didn't raise the assertion. Thanks again for your help, I will close this issue.
